### ISSUE TITLE
YouTube Disabled the Addon again

### ISSUE THREAD

**Issue by McBobb** [feller-prj/extractor-project#47]
First and foremost, I need to express my appreciation and gratitude at how quickly you manage to update this addon with every change YouTube makes.

As of Tuesday, Jan 11, 2022, The button to open media to download selector window is no longer present on YouTube videos.

I hope that you're able to work on this at your earliest convenience.

And thanks again.

**Comment by fer-go-nan** [comment]
Hello. I'm not sure if it's the same problem. 
In Firefox, when I try to download a youtube video, the sand clock appears spinning and after a couple of seconds it freezes. Later a warning pops up about a page slowing down Firefox.
Thanks for your work and time.

**Comment by negusm** [comment]
> Hello. I'm not sure if it's the same problem. In Firefox, when I try to download a youtube video, the sand clock appears spinning and after a couple of seconds it freezes. Later a warning pops up about a page slowing down Firefox. Thanks for your work and time.

I get this behavior as well.  But it takes more than a few seconds for the warning.

**Comment by LogicDeLuxe** [comment]
I encountered the freezing issue as well. After the today's update, the freezing is gone, but the hourglass just keeps spinning forever. I'm on Firefox for Linux Mint 95.0.1 (64-bit).

**Comment by Dentrox** [comment]
Hi, i'm using your addon and its very useful. I have the same issue reportes here. The sandclock stay rotating forever.
Thank you in advance!

**Comment by negusm** [comment]
Tested in Opera and it fails with:  "Cannot detect url_encoded_fmt_stream_map or adaptive_fmts"

**Comment by Simon-J-N-Reading** [comment]
I think it may be harder to fix this time...
"Content Security Policy: The page’s settings blocked the loading of a resource at blob:moz-extension://24dd0d08-5f49-4bac-8a8d-1180e12d2502/78440136-522d-4241-af38-7787c61732dc (“script-src”)."
Injected scripts have been effectively blocked :- therefore I am not sure what workarounds are possible with the existing architecture.

**Comment by Draghorn** [comment]
Reviwers on the add-on page have discovered previous versions of the plugin don't have the freezing problem.  I would go with version 1.1.7.2.  When I tried version 1.1.6 I was getting the extremely slow download speeds (below 100 KB/sec).  1.1.7.2 seems to not be freezing and actually uses some decent portion of your actual bandwidth.

**Comment by McBobb** [comment]
The button has returned to YouTube video pages, but while clicking it does open the available video version list window, the window never populates. It just has an endless spinning hourglass icon.

Win 10 Home
Firefox 96.0.1
Addon Version 1.1.8

Not sure how to check Native Client version number, but the addon says it's the most recent.

**Comment by CodingMarkus** [comment]
> Reviwers on the add-on page have discovered previous versions of the plugin don't have the freezing problem. I would go with version 1.1.7.2. 

I can confirm that 1.1.7.2 just works fine for me in Firefox 96 on macOS 11.

Whoever is interested in switching to that version, just go here and install it. It will override the current one and you can download again:

https://addons.mozilla.org/en-US/firefox/addon/youtube_downloader_webx/versions/

**Comment by TheGhost78** [comment]
> > Reviwers on the add-on page have discovered previous versions of the plugin don't have the freezing problem. I would go with version 1.1.7.2.
> 
> I can confirm that 1.1.7.2 just works fine for me in Firefox 96 on macOS 11.
> 
> Whoever is interested in switching to that version, just go here and install it. It will override the current one and you can download again:
> 
> https://addons.mozilla.org/en-US/firefox/addon/youtube_downloader_webx/versions/

Confirmed for Firefox 96.0.1 in Windows 10 Pro x64.

**Comment by McBobb** [comment]
Still endless hourglass on Firefox 96.0.1 in Windows 10 Home x64

**Comment by IceC00l** [comment]
I also can confirm that downgrading to 1.1.7.2 works on Firefox 96.0.1 and WIn 10 Pro 64 (19044.1466).

**Comment by McBobb** [comment]
Downgrading to 1.1.7.2 on Win 10 Home x64 returns to the problem of the icon to open the addon window on YouTube videos is missing.

**Comment by McBobb** [comment]
Furthermore, after downgrading to 1.1.7.2, then upgrading back to the most current version, 1.1.8, the button is missing from YouTube pages in that version as well!

**Comment by CodingMarkus** [comment]
> Furthermore, after downgrading to 1.1.7.2, then upgrading back to the most current version, 1.1.8, the button is missing from YouTube pages in that version as well!

The missing button is caused by a recent change at YouTube. Today I also miss the button in 1.1.7.2, but until two days ago the button was still there using that version.

**Comment by Draghorn** [comment]
Yep, I can also confirm that the embedded icon has gone missing on youtube pages in the last few days.  Thankfully, this addon also has a toolbar icon you can use to bring up the download interface.  You should still be getting a yD button up in your address/toolbar.  You can click on that and then "Download Panel" to bring up files to download.

> Furthermore, after downgrading to 1.1.7.2, then upgrading back to the most current version, 1.1.8, the button is missing from YouTube pages in that version as well!

So, I wouldn't upgrade back to the current version until someone can confirm that it will populate the download panel correctly, or possibly a new version comes out to fix the current's download panel problem.  I should note that by default Firefox will try to upgrade you to the current version.  If you want to stay on an older version you need to bring up the extension's information on your addons page, and then set it's automatic updates to "off".

**Comment by IceC00l** [comment]
Exactly, the youtube icon has disappeared _but this doesn't prevent the functionality of the add-on_ as you can access the download panel from the "YD" icon on the browser toolbar and from there do what you need - so until a new version has confirmed the fix stay on 1.1.7.2.

**Comment by McBobb** [comment]
Thanks for the suggestions. I've downgraded back to 1.1.7.2 and will see if the toolbar button works the next time I want to DL a YT vid.

Something the creator might want to look at is this addon

https://addons.mozilla.org/en-CA/firefox/addon/youtube-video-easy/

It also currently works on YouTube Pages, and is able to place its download button right on the body of the YouTube page still. Honestly, I'm loyal to this addon creator, and prefer his addon because he's shown such dedication to keeping it up to date with YouTube nuttiness throughout the years. But maybe he can look at how that addon is managing to place it's "open the menu" button on YT pages and copy it?

**Comment by Vvoids** [comment]
Still not possible to download anymore*

**Comment by JarieSuisen** [comment]
Reverting to 1.1.7.2 isn't working for me. It is now able to bring up the download overlay, but attempting to download by any means (even quick download) just causes it to become unresponsive and then the whole tab dies.

**Comment by CodingMarkus** [comment]
> Reverting to 1.1.7.2 isn't working for me. 

That's because you are a few days late to the party. The actual download only broke recently. It seems like YouTube is currently making change after change to keep breaking current or past versions of the addon. As I know of other addons that still work, it seems like YouTube is primary focusing this addon at the moment or the other ones are better at quickly adapting to these changes.

**Comment by Simon-J-N-Reading** [comment]
@CodingMarkus  
Youtube has introduced a CSP (content security policy) meta like "<meta http-equiv="Content-Security-Policy" content="default-src *; style-src 'self' 'unsafe-inline'; script-src 'self' ' etc ...... > - in other words : "Dear browser - Only parse the JavaScript we give you" - to protect their page from third party injected JavaScript  (As I alluded to in a previous comment) 
It is not targeted at this specific Addon - It is targeting all Addons that inject scripts - (Like those to remove adverts , Alter the way the video is played when embedded etc..)
This Addon will, I suspect, need  a complete change in approach before it will function again.

**Comment by MM-John** [comment]
@McBobb

> Something the creator might want to look at is this addon
> 
> https://addons.mozilla.org/en-CA/firefox/addon/youtube-video-easy/

This did not work for me. No button. Clicking the toolbar icon goes to this URL which times out:

hxxps://www.ultimatevideodownloader[dot]com/#url=https://www.youtube.com/watch?v=EXQGTInPpZU&list=PLcHqC9YNzSaB621igMmxfeMUvroQqYr3L&index=3

On Windows 11 (not 10, 11) 21H2 firefox 96.0.2

**Comment by MM-John** [comment]
This page 

https://addons.mozilla.org/en-US/firefox/addon/youtube_downloader_webx/

Now lists 1.1.7.2 as its latest version

>Version
    1.1.7.2
Size
    233 KB
Last updated
    a month ago (Dec 20, 2021)

I had installed it from here a few minutes ago and the "Manage extension" reported itself as 1.1.8. I wonder if the developer just now, 12:00pm eastern 01/26/2022 reverted to the old version?

Also the version page:

https://addons.mozilla.org/en-US/firefox/addon/youtube_downloader_webx/versions/

Lists 1.1.7.2 as the lastest version at the top. It had a 'Remove' button for me. I guess it has a 'Remove' button at the top for the latest version no matter what version you have installed?

Anyway reinstalling from the main page now gives me 1.1.7.2.

Unfortunately, it still freezes in youtube for me on 1.1.7.2. It was doing the spinning hourglass for me on 1.1.8.

1.1.6 worked for me.

Windows 11 (not 10) 21H2
Firefox 96.0.2

**Comment by Draghorn** [comment]
Yep, 1.1.8 seems to have been pulled.  And while 1.1.6 technically works, you will be stuck on extremely slow speeds (basically at or below 100 KB/sec).  Only really usable on the tiniest of videos.  I've gone to an alternative addon even though it is limited to 720p.  I've been having two youtube downloader addons installed so they can be a backup for each other, cause you never know when Google will strike and make your addons break.

**Comment by McBobb** [comment]
> @McBobb
> 
> > Something the creator might want to look at is this addon
> > https://addons.mozilla.org/en-CA/firefox/addon/youtube-video-easy/
> 
> This did not work for me. No button. Clicking the toolbar icon goes to this URL which times out:
> 
> hxxps://www.ultimatevideodownloader[dot]com/#url=https://www.youtube.com/watch?v=EXQGTInPpZU&list=PLcHqC9YNzSaB621igMmxfeMUvroQqYr3L&index=3
> 
> On Windows 11 (not 10, 11) 21H2 firefox 96.0.2

It still works fine for me, while even version 1.1.7.2. still gives me an endless spinning hourglass.

But the point I was saying is that addon still manages to get its download button to appear on the most current version of YouTube, while Feller's addon still cannot. That's specifically what I was mentioning it for.

**Comment by CodingMarkus** [comment]
> It is not targeted at this specific Addon - It is targeting all Addons that inject scripts - (Like those to remove adverts , Alter the way the video is played when embedded etc..) 

However, that raises the question why "[Easy Youtube Video Downloader Express](https://addons.mozilla.org/en-US/firefox/addon/easy-youtube-video-download)" still works as it has been working before (including the injected buttons on the site) and why "[AdBlocker for YouTube](https://addons.mozilla.org/en-US/firefox/addon/adblock-for-youtube)" still reliably blocks all adds. Apparently those can do all of this without having to inject any scripts or they can inject them without being affected? 🤷‍♂️

**Comment by Simon-J-N-Reading** [comment]
@CodingMarkus 
As you will see in my first post - I copied the error : I did not rely on suppositions , but investigation.

**Comment by MasterZeroFlash** [comment]
Any updates on this issue? Easy Youtube Video Downloader Express indeed works (and with good speeds), but I dont feel like paying for the Pro version (and I've seen copy-pasted ads for the above addon in the reviews of other similar addons, also their justification for charging money is that apparently high quality (HD) isnt available from Youtube itself via direct links anymore, so in their own words it takes a lot of server infrastructure to provide such a service, but Im not sure I buy that argument).

Maybe that addon somehow tricks Youtube into thinking that it is actually playing a video on a screen when in reality it is downloading it, and thus the proper download speeds (because I'm pretty sure 50kb/s isnt enough to even keep the audio going in a properly played video clip XD (50kb/s being the speed I get when downgrading this projects addon to a version that works).

Im no expert by any means though, but I'd like to think that the solution is somehow pretty simple, and the thing about it rejecting scripts doesnt really cut it when one clearly works (at least the free version) and the other barely works.

**Comment by Simon-J-N-Reading** [comment]
@MasterZeroFlash :   The error I quoted was generated by this Add on -  It is the actual fail point  : so it does cut it (very precisely :-) ) , Easy Youtube Video Downloader Express does not inject external scripts .  - so therefore it does not generate this type of error.
(I have delved into how Easy Youtube Video Downloader works,  and , if you are interested in coding there is some smart stuff - But this is not the place for  bringing up others code)
The developer of this Add on did a great job for ages (Kudos) - and made it available FOC - (double Kudos)  I hope it comes back good and strong, And if I have provided a tiny bit of useful debugging to the developer : that's great.

**Comment by MasterZeroFlash** [comment]
> @MasterZeroFlash : The error I quoted was generated by this Add on - It is the actual fail point : so it does cut it (very precisely :-) ) , Easy Youtube Video Downloader Express does not inject external scripts . - so therefore it does not generate this type of error. (I have delved into how Easy Youtube Video Downloader works, and , if you are interested in coding there is some smart stuff - But this is not the place for bringing up others code) The developer of this Add on did a great job for ages (Kudos) - and made it available FOC - (double Kudos) I hope it comes back good and strong, And if I have provided a tiny bit of useful debugging to the developer : that's great.

*Nods*, Fair enough, I took your comment as more of a blanket statement than it was meant to be.
